driver-core/src/main/com/mongodb/internal/operation/ChangeStreamBatchCursor.java

```diff
@@ -41,22 +43,31 @@ final class ChangeStreamBatchCursor<T> implements AggregateResponseBatchCursor<T
     private final ReadBinding binding;
     private final ChangeStreamOperation<T> changeStreamOperation;
     private final int maxWireVersion;
-
+    private final TimeoutContext timeoutContext;
     private CommandBatchCursor<RawBsonDocument> wrapped;
     private BsonDocument resumeToken;
     private final AtomicBoolean closed;
 
+    /**
+     * This flag is used to manage change stream resumption logic after a timeout error.
+     * Indicates whether the last {@code next} call resulted in a {@link MongoOperationTimeoutException}. If {@code true},
+     * indicates a timeout occurred, prompting an attempt to resume the change stream on the next call.
+     */
+    private boolean lastOpTimeoutStatus;
```

Comment:
The naming / use of this flag is confusing. Can you clarify in the code / docs that throwing a `MongoOperationTimeoutException` doesn't invalidate the cursor.

The reason for this is in 6 months we'll all have forgotten about this special and strange CSOT behavior.

Reply:
Acknowledged. I have renamed the field and added additional class level javaDoc. Please let me know if there is anything to add.